### PR TITLE
update SwiftLintFramework/Extensions to follow Swift 3 API Design Guidelines

### PR DESCRIPTION
**Pull request by jpsim**
1 out of 7 Swift 3 API Guidelines PRs. /cc @marcelofabri

### CODE REVIEW

File: Source/SwiftLintFramework/Extensions/File+SwiftLint.swift [hunk 1]
```diff
@@ -153,29 +144,28 @@ extension File {
...
      file contents.
      */
-    internal func matchPattern(_ pattern: String,
-                               excludingSyntaxKinds syntaxKinds: [SyntaxKind],
-                               range: NSRange? = nil) -> [NSRange] {
-        return matchPattern(pattern, range: range).filter {
-            $0.1.filter(syntaxKinds.contains).isEmpty
-        }.map { $0.0 }
+    internal func match(pattern: String,
+                        excludingSyntaxKinds syntaxKinds: [SyntaxKind],
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/File+SwiftLint.swift]: I feel that this is not consistent with the other signature that uses `with`.

File: Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift [hunk 2]
```diff
@@ -9,11 +9,11 @@
 import Foundation
 
 public protocol LintableFileManager {
-    func filesToLintAtPath(_ path: String, rootDirectory: String?) -> [String]
+    func filesToLint(inPath: String, rootDirectory: String?) -> [String]
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: `filesToLint(in:, rootDirectory:)`?

File: Source/SwiftLintFramework/Rules/ClosingBraceRule.swift [hunk 3]
```diff
@@ -13,8 +13,7 @@ private let whitespaceAndNewlineCharacterSet = CharacterSet.whitespacesAndNewlin
 
 extension File {
     fileprivate func violatingClosingBraceRanges() -> [NSRange] {
-        return matchPattern(
-            "(\\}[ \\t]+\\))",
+        return match(pattern: "(\\}[ \\t]+\\))",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/ClosingBraceRule.swift]: vertical param alignment

File: Source/SwiftLintFramework/Rules/ClosingBraceRule.swift [hunk 4]
```diff
@@ -53,9 +52,9 @@ public struct ClosingBraceRule: CorrectableRule, ConfigurationProviderRule {
     }
 
     public func correctFile(_ file: File) -> [Correction] {
-        let violatingRanges = file.ruleEnabledViolatingRanges(
+        let violatingRanges = file.ruleEnabled(violatingRanges:
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/ClosingBraceRule.swift]: shouldn't the param name be in the same line as the value?

File: Source/SwiftLintFramework/Rules/OpeningBraceRule.swift [hunk 5]
```diff
@@ -13,8 +13,7 @@ private let whitespaceAndNewlineCharacterSet = CharacterSet.whitespacesAndNewlin
 
 extension File {
     fileprivate func violatingOpeningBraceRanges() -> [NSRange] {
-        return matchPattern(
-            "((?:[^( ]|[\\s(][\\s]+)\\{)",
+        return match(pattern: "((?:[^( ]|[\\s(][\\s]+)\\{)",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/OpeningBraceRule.swift]: parameters vertical alignment

File: Source/SwiftLintFramework/Rules/OpeningBraceRule.swift [hunk 6]
```diff
@@ -77,9 +76,9 @@ public struct OpeningBraceRule: CorrectableRule, ConfigurationProviderRule {
     }
 
     public func correctFile(_ file: File) -> [Correction] {
-        let violatingRanges = file.ruleEnabledViolatingRanges(
+        let violatingRanges = file.ruleEnabled(violatingRanges:
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/OpeningBraceRule.swift]: param name should be in same line as value (?)

File: Source/SwiftLintFramework/Rules/RedundantNilCoalescingRule.swift [hunk 7]
```diff
@@ -11,10 +11,8 @@ import SourceKittenFramework
 
 extension File {
     fileprivate func violatingRedundantNilCoalescingRanges() -> [NSRange] {
-        return matchPattern(
-            "\\s*\\?\\?\\s*nil\\b", // {whitespace} ?? {whitespace} nil {word boundary}
-            withSyntaxKinds: [.keyword]
-        )
+        return match(pattern: "\\s*\\?\\?\\s*nil\\b", // {whitespace} ?? {whitespace} nil {word boundary}
+            with: [.keyword])
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/RedundantNilCoalescingRule.swift]: alignment

File: Source/SwiftLintFramework/Rules/SortedImportsRule.swift [hunk 8]
```diff
@@ -27,8 +27,8 @@ public struct SortedImportsRule: ConfigurationProviderRule, OptInRule {
     )
 
     public func validateFile(_ file: File) -> [StyleViolation] {
-        let importRanges = file.matchPattern("import\\s+\\w+",
-                                             withSyntaxKinds: [.keyword, .identifier])
+        let importRanges = file.match(pattern: "import\\s+\\w+",
+                                             with: [.keyword, .identifier])
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/SortedImportsRule.swift]: alignment

File: Source/SwiftLintFramework/Rules/TrailingSemicolonRule.swift [hunk 9]
```diff
@@ -52,9 +52,9 @@ public struct TrailingSemicolonRule: CorrectableRule, ConfigurationProviderRule
     }
 
     public func correctFile(_ file: File) -> [Correction] {
-        let violatingRanges = file.ruleEnabledViolatingRanges(
+        let violatingRanges = file.ruleEnabled(violatingRanges:
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/TrailingSemicolonRule.swift]: param name in same line as value (?)

File: Source/SwiftLintFramework/Extensions/File+SwiftLint.swift (hunk 1 above)
Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/File+SwiftLint.swift]: hehe, that's changed in another PR I think. Yay splitting these up?

File: Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift (hunk 2 above)
Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: if the parameter was a `URL` type, that'd be more obvious. But as it stands, the string could be anything, so the API specifies what the string is used for.

Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: ok, makes sense 👍

File: Source/SwiftLintFramework/Extensions/SyntaxMap+SwiftLint.swift [hunk 10]
```diff
@@ -13,11 +13,11 @@ extension SyntaxMap {
     /// Returns array of SyntaxTokens intersecting with byte range
     ///
     /// - Parameter byteRange: byte based NSRange
-    internal func tokensIn(_ byteRange: NSRange) -> [SyntaxToken] {
+    internal func tokens(inByteRange byteRange: NSRange) -> [SyntaxToken] {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Extensions/SyntaxMap+SwiftLint.swift]: I like how this now express that it's a byte range - I've always had to lookup in the header

File: Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift (hunk 2 above)
Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: See the "Compensate for weak type information" section of the guidelines:

image

Comment (jpsim) [inline Source/SwiftLintFramework/Extensions/NSFileManager+SwiftLint.swift]: https://swift.org/documentation/api-design-guidelines/

**Pull request merged**